### PR TITLE
[IMAGE-BUTTON][VM|DS] Add selected and error state to image button

### PR DESCRIPTION
**Pull request by v1shva**
…tyling.

**Changes**
Description of changes...

- delete branch

 - Add selected state to image button
 - Add error state to image button
 - Update styling

**Additional information**

- You may refer to this  https://www.figma.com/file/us90jOpWBahsqK2VAluPvD/BookingSG-Design-System?type=design&node-id=9405-80023&t=2LQKtkAuY0dRaFcq-4
- Jira : https://jira.ship.gov.sg/browse/RBS-400

### CODE REVIEW

File: src/image-button/image-button.style.tsx [hunk 1]
```diff
@@ -21,29 +21,31 @@ export const Main = styled.button`
...
-
-        @media (hover: hover) {
-            &:hover {
-                box-shadow: 0px 8px 16px -8px ${Color.Shadow.Elevation};
-                ${TextStyleHelper.getTextStyle("H4", "bold")}
-            }
-        }
+    &:hover {
+        border: 1px solid ${Color.Accent.Light[1]};
+        box-shadow: ${DesignToken.InputBoxShadow};
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:33]: input box shadows are inset, follow Toggle's instead

File: src/image-button/image-button.style.tsx [hunk 2]
```diff
@@ -21,29 +21,31 @@ export const Main = styled.button`
 
     img {
         width: 3.5rem;
-    }
-
-    ${MediaQuery.MaxWidth.tablet} {
-        padding: 1.5rem 0rem;
+        height: 3.5rem;
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:24]: can add a object-fit `contain` as a fallback for different aspect ratios

File: src/image-button/image-button.style.tsx [hunk 3]
```diff
@@ -21,29 +21,31 @@ export const Main = styled.button`
...
+    &:hover {
+        border: 1px solid ${Color.Accent.Light[1]};
+        box-shadow: ${DesignToken.InputBoxShadow};
+        ${TextStyleHelper.getTextStyle("H4", "bold")}
     }
+    ${(props) => {
+        if (props.$selected) {
+            return css`
+                background: ${Color.Accent.Light[5]};
+                border-color: ${Color.Primary};
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:40]: missing the border width

File: src/image-button/image-button.style.tsx [hunk 4]
```diff
@@ -21,29 +21,31 @@ export const Main = styled.button`
...
+                color: ${Color.Primary(props)};
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:41]:
```suggestion
                color: ${Color.Primary};
```

File: src/image-button/image-button.style.tsx [hunk 5]
```diff
@@ -21,29 +21,31 @@ export const Main = styled.button`
...
+        padding: 0.5rem 0.25rem;
     }
 
-    ${MediaQuery.MinWidth.tablet} {
-        :focus:active {
-            box-shadow: ${DesignToken.ElevationBoxShadow};
-            ${TextStyleHelper.getTextStyle("H4", "bold")}
-        }
-
-        @media (hover: hover) {
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:40]: is this media query still needed?

**Review by qroll — CHANGES_REQUESTED** [review]
- for mobile, `Padding between text and image` is written as 4px but the mockup uses 8px. which is the intended one?
- is there a disabled + selected state?

File: src/image-button/image-button.style.tsx (hunk 1 above)
Comment (v1shva) [reply src/image-button/image-button.style.tsx:33]: sure, Padding between text and image should be 4px for mobile, there's no selected + disabled state.

File: src/image-button/image-button.style.tsx (hunk 5 above)
Comment (v1shva) [reply src/image-button/image-button.style.tsx:40]: removed.

Comment (qroll) [reply src/image-button/image-button.style.tsx:40]: i believe long presses do trigger hover and tend to "stick" if the user doesn't explicitly tap on something else, but it's purely a cosmetic issue

File: src/image-button/image-button.style.tsx [hunk 6]
```diff
@@ -1,10 +1,10 @@
 import { MediaQuery } from "../media";
-import styled from "styled-components";
+import styled, { css } from "styled-components";
 import { Color } from "../color/color";
 import { DesignToken } from "../design-token/design-token";
 import { TextStyleHelper } from "../text/helper";
 
-export const Main = styled.button`
+export const Main = styled.button<{ $selected?: boolean; $error?: boolean }>`
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:7]: can add a transparent 1px border to keep the height consistent

File: src/image-button/image-button.style.tsx [hunk 7]
```diff
@@ -21,29 +21,45 @@ export const Main = styled.button`
...
+                    border: 1px solid ${Color.Accent.Light[1]};
+                    box-shadow: 0 0 4px 1px ${Color.Shadow.Accent};
+                    ${TextStyleHelper.getTextStyle("H4", "bold")}
+                }
+            `;
         }
-    }
+    }}
 
     :disabled {
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:64]: set border to transparent here as well

File: src/image-button/image-button.style.tsx [hunk 8]
```diff
@@ -21,29 +21,45 @@ export const Main = styled.button`
...
+        gap: 0.25rem;
     }
 
-    ${MediaQuery.MinWidth.tablet} {
-        :focus:active {
-            box-shadow: ${DesignToken.ElevationBoxShadow};
-            ${TextStyleHelper.getTextStyle("H4", "bold")}
-        }
+    ${(props) => {
+        if (props.$error) {
```
Comment (qroll) [inline src/image-button/image-button.style.tsx:34]: is there a selected + error styling also?

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/image-button/image-button.style.tsx (hunk 8 above)
Comment (v1shva) [reply src/image-button/image-button.style.tsx:36]: yes, there can be, updated the PR.

File: src/image-button/image-button.style.tsx (hunk 7 above)
Comment (qroll) [reply src/image-button/image-button.style.tsx:65]: should have specified. this is intended for the disabled style to remove the box-shadow on hover

Comment (v1shva) [reply src/image-button/image-button.style.tsx:64]: ah, my bad, didn't check if disabled state had any hover effects. fixed now.

**Review by qroll — APPROVED** [review]

**Pull request merged**